presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedHistogram.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+
+public class TestTypedHistogram
+{
+    @Test
+    public void testMassive()
+        throws Exception
+    {
+        TypedHistogram typedHistogram = new TypedHistogram(BIGINT, 1000);
+        List<Object> input = IntStream.range(1, 2000).mapToObj(i -> IntStream.iterate(i, j -> j).limit(i).mapToObj(k -> k)).flatMap(l -> l).collect(Collectors.toList());
```

Comment:
Additionally, you don't need to collect them into a list and then loop over the list. You can do `forEach` on the stream.